### DISCUSSION TITLE
Accessibility check support

### DISCUSSION THREAD

**Discussion by arvigeus** [epicweb-dev/epic-stack#45] [Ideas]
The idea is to allow the developer to check if their app is accessible.

Cons:

- Complex to implement
- Annoying, if the dev does not care (example: personal projects)

**Comment by kentcdodds** [comment]
Do you have tool suggestions?

**Reply by arvigeus** [reply]
Playwright has [Axe integration](https://playwright.dev/docs/accessibility-testing). There is also [eslint plugin](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) for static checks, but could be too annoying if people don't want it.

Another tool, while not strictly for accessibility only, is this unofficial [lighthouse plugin](https://www.npmjs.com/package/playwright-lighthouse)  for Playwright.

**Reply by kentcdodds** [reply]
I'd be willing to look at a PR that adds Axe integration in the Playwright tests.

**Reply by abhinaba-ghosh** [reply]
You could into here - https://github.com/abhinaba-ghosh/axe-playwright

@arvigeus I haven't created the lighthouse plugin for accesibility testing in focus

**Reply by andrecasal** [reply]
Axe looks great btw. And testing seems like the right place to put this 👍

**Reply by andrecasal** [reply]
Looking at https://github.com/abhinaba-ghosh/axe-playwright for a PR for this.

Seems like we have to visit a page, inject the Axe script, and check for accessibility: `page.goto()` -> `injectAxe(page)` -> `checkA11y()`. For every page.

Is there a way to get all Remix routes so we can check all pages?

**Reply by kentcdodds** [reply]
You could implement a sitemap and then use that. There's an example for this.

**Reply by arvigeus** [reply]
@abhinaba-ghosh What is the difference between this and @axe-core/playwright? It seems to be the official way to do it
